### DISCUSSION TITLE
OTP integration

### DISCUSSION THREAD

**Discussion by RamonGal** [wg-easy/wg-easy#659] [Ideas]
A seemingly weak part of this configuration is that someone can forcibly try to break the password, or if the password gets leaked somehow, the entire system would be in danger.

One way to take that into consideration is to use mfa like otp to enable a further layer of security.

What are your thoughts on this?
Is it something you could see integrating into the project as an add-on?

Thanks

**Comment by ManuProd** [comment]
I agree with you, for a VPN Web UI, MFA should be mandatory even if wg-easy is great.

**Comment by RamonGal** [comment]
I've been wanting to do something similar to this for a client and could make a PR to the project if the creator wants to. @WeeJeWel

**Reply by WeeJeWel** [reply]
Honestly I think it's better to outsource authentication entirely to another tool, e.g. nginx as reverse proxy.

I think that the Linux philosophy of 'one tool one job' is still great and should also apply to web projects.

**Comment by mpfl** [comment]
I've done this by disabling wg-easy's built in authentication and hiding wg-easy behind Authelia with Traefik. In Authelia, only members of a certain group can access the wg-easy web UI.

**Comment by peterlewis** [comment]
This sounds like a sensible way forward! I.e. keep wg-easy auth as no-frills as possible (like it is currently) with a further exposed environment variable (or potentially when `$PASSWORD` variable is not populated) to disable auth, when it's being used with external auth provider.
